src/Shared/ServerCertificateValidationHandler.cs

```diff
@@ -6,35 +6,36 @@
 using System;
 using System.Net.Http;
 
-namespace OpenTelemetry.ResourceDetectors.AWS.Http;
+namespace OpenTelemetry.ResourceDetectors;
 
-internal class Handler
+internal static class ServerCertificateValidationHandler
 {
-    public static HttpClientHandler? Create(string certificateFile)
+    public static HttpClientHandler? Create(string certificateFile, IServerCertificateValidationEventSource log)
     {
         try
         {
-            ServerCertificateValidationProvider? serverCertificateValidationProvider =
-                ServerCertificateValidationProvider.FromCertificateFile(certificateFile);
+            ServerCertificateValidationProvider? serverCertificateValidationProvider = ServerCertificateValidationProvider.FromCertificateFile(certificateFile, log);
 
             if (serverCertificateValidationProvider == null)
             {
-                AWSResourcesEventSource.Log.FailedToValidateCertificate(nameof(Handler), "Failed to Load the certificate file into trusted collection");
-                return null;
+                return new HttpClientHandler();
```

Comment:
This seems like a behavior change - why remove the log statement, and change the `return null` to `return new`?

Reply:
It is, but should not really affect anything. All required log information would be logged inside call `ServerCertificateValidationProvider.FromCertificateFile`. If we were unable to create handler with specific policy to trust the certificate, we still create default handler that will use system-wide SSL trust to establish connection. It is OK, as we usually want to trust additional certificate, not limit anything (at least it is how custom ServerCertificateValidationProvider is implemented).
Most probably the connection would fail to be established, but it would also be caught and logged later - with reasonable information that SSL connection was not established. With prior messages about failure to create validation provider, entire picture would be clear.